qiita_pet/handlers/api_proxy/processing.py

```diff
@@ -80,9 +83,30 @@ def list_options_handler_get_req(command_id):
          'options': list of dicts of {'id: str', 'name': str,
                                       'values': dict of {str: str}}}
     """
+    def _helper_process_params(params):
+        return dumps(
+            {k: str(v).lower() for k, v in params.items()}, sort_keys=True)
+
     command = Command(command_id)
+    rparamers = command.required_parameters.keys()
+    eparams = []
+    if artifact_id is not None:
+        artifact = Artifact(artifact_id)
+        for job in artifact.jobs(cmd=command):
+            jstatus = job.status
+            outputs = job.outputs if job.status == 'success' else None
+            # this ignore any jobs that weren't successful or are in
+            # construction, or the results have been deleted [outputs == {}]
+            if jstatus not in {'success', 'in_construction'} or outputs == {}:
+                continue
+            params = job.parameters.values
```

Comment:
Is this a copy or the same object? I'm wondering if the `del` below will modify state in `job.parameters.values`

Reply:
Nop, different ... but I can easily add a .copy if you think is better, should I?

Reply:
That would be defensive, thanks